src/profile/providers/SearchFilter.tsx

```diff
@@ -36,43 +35,21 @@ const INITIAL_STATE: IState = {
 
 export const ProfileSearchContext = createContext(INITIAL_STATE);
 
-const parseRange = (rangeString: string): [number, number] | null => {
-  const match = rangeString.match(/^\[\d\,\d\]$/);
-  if (!match) {
-    return null;
-  }
-  const [n, m] = match.map((k) => Number(k));
-  return [n, m];
-};
-
 export const ProfileSearchProvider: FC = ({ children }) => {
   const { user } = useContext(UserContext);
   /** Should not be able to render this page without an authenticated user */
   if (!user) {
     return null;
   }
 
-  /** Store filter parameters in browser query */
-  const [querySearch, setSearch] = useQueryParam('search');
-  const [queryGroup, setQueryGroup] = useQueryParam('group');
-  const [queryRange, setQueryRange] = useQueryParam('range');
+  const [search, setSearch] = useState<string | undefined>(undefined);
+  const [group, setGroup] = useState<string | undefined>(undefined);
+  const [range, setRange] = useState<[number, number]>([1, 6]);
 
   const [profiles, setProfiles] = useState<IPublicProfile[]>([]);
   /** Restrict possibility of calling nextPage until the first page of current query has been added to state. */
   const [ready, setReady] = useState(false);
 
-  const search = querySearch || undefined;
-  const group = queryGroup || undefined;
-  const range = parseRange(queryRange || '') || [1, 6];
-  const setGroup = (newGroup: string) => {
-    if (newGroup.toLowerCase() === 'alle grupper') {
-      setQueryGroup(null);
-    } else {
-      setQueryGroup(newGroup);
-    }
-  };
-  const setRange = (newRange: [number, number]) => setQueryRange(JSON.stringify(newRange));
```

Comment:
I have to say this part was confusing as it was previously, but there is no problem in removing it? What did it do?

Reply:
We had the state of search filters stored as strings in the url, but i removed it because I found it too complicated to work with. I'm doing this mostly to make things easier to maintain in the long run. Since it used internal React Router things anyways, I decided it was not worth it to re-write it for the Next Router